File: 02-week-2--recursion/05-day-5--recursive-selection-sort/ruby/selection_sort_recursive.rb
def selection_sort_recursive(arr)
  # type your code in here
  return [] if arr.length == 0 #base case

  min_value = arr.min #splat operator 
  index_value = arr.index(min_value)
  arr.delete_at(index_value)
  

  result = selection_sort_recursive(arr)
  result.unshift(min_value)

   
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort_recursive([3, -1, 5, 2])

  puts

  puts "Expecting: [-8, -2, 4, 6]"
  print "=> "
  print selection_sort_recursive([4, -8, -2, 6])
  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 02-week-2--recursion/05-day-5--recursive-selection-sort/javascript/selection_sort_recursive.js
function selectionSortRecursive(arr) {
  // type your code here
 if (arr.length === 0) return [] //base case

 const minValue = Math.min(...arr)
 const index = arr.indexOf(minValue)
 arr.splice(index, 1)

 const result = selectionSortRecursive(arr)
 result.unshift(minValue)

 return result 
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSortRecursive([3, -1, 5, 2]));

  console.log("");
}

module.exports = selectionSortRecursive;

// Please add your pseudocode to this file
// And a written explanation of your solution

//Iterative Solution

//function selection_sort_recursive(arr){

//for(let i = 0; i < arr.length - 1; i++){

  // let minIndex = i
  // for (let j = i + 1; j < arr.length; j++){

    // if( arr[j] < arr[minIndex]){

      // minIndex = j
    // }
  // }
  // [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]]
// }
  // return arr
//}


